### PR TITLE
feat(analysis): add provider and source filtering to analysis requests

### PR DESCRIPTION
**Pull request by ruromero**
## Summary
- Add `--providers` and `--sources` CLI flags to `stack`, `component`, and `stack-batch` commands
- Add `TRUSTIFY_DA_PROVIDERS` and `TRUSTIFY_DA_SOURCES` env var fallbacks
- Extract shared `appendAnalysisQueryParams()` helper in `analysis.js`
- 9 new tests covering URL construction and HTTP integration
- Add `--providers` and `--sources` flags to `image` command for consistency ([TC-5441](https://redhat.atlassian.net/browse/TC-5441))

Implements [TC-5409](https://redhat.atlassian.net/browse/TC-5409)

## Test plan
- [x] Unit tests verify URL construction with providers/sources params
- [x] Unit tests verify env var fallback
- [x] Integration tests verify params forwarded to mock backend
- [x] `npm run lint` passes
- [x] `npm test` passes

[TC-5441]: https://redhat.atlassian.net/browse/TC-5441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5409]: https://redhat.atlassian.net/browse/TC-5409?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by a-oren — COMMENTED** [review]
image CLI command is missing --providers / --sources flags.

The requestImages function in analysis.js correctly calls appendAnalysisQueryParams, but the image CLI command doesn't expose --providers or --sources options. The three other commands (component, stack, stack-batch) all got the flags. This means image analysis supports filtering via env vars and the programmatic API, but not via the CLI, which is inconsistent.

File: src/analysis.js [hunk 1]
```diff
@@ -247,6 +239,25 @@ async function requestImages(imageRefs, url, html = false, opts = {}) {
...
+	}
+	const providers = getCustom('TRUSTIFY_DA_PROVIDERS', null, opts);
+	if (providers) {
+		url.searchParams.append('providers', providers);
+	}
+	const sources = getCustom('TRUSTIFY_DA_SOURCES', null, opts);
+	if (sources) {
+		url.searchParams.append('sources', sources);
+	}
+}
```
[the remark this answers is not in the document]
Comment (ruromero) [reply src/analysis.js:259]: [sdlc-workflow/verify-pr] Classified as **suggestion** — proposes using `set` instead of `append` for URLSearchParams to prevent potential duplicates. The function is called once per URL creation in each request function, so duplicates cannot occur in current usage. CONVENTIONS.md does not document a preference, and the codebase uses `append` exclusively. No sub-task created.

File: test/analysis.test.js [hunk 2]
```diff
@@ -207,6 +207,140 @@ suite('testing the analysis module for sending api requests', () => {
...
+			expect(url.searchParams.get('providers')).to.equal('redhat')
+			expect(url.searchParams.get('sources')).to.equal('osv,nvd')
+		})
+
+		/** Verifies that no providers or sources params are added when neither is set. */
+		test('does not append providers or sources when neither is set', () => {
+			const url = new URL('http://example.com/api/v5/analysis')
+			analysis.appendAnalysisQueryParams(url, {})
+			expect(url.searchParams.has('providers')).to.equal(false)
+			expect(url.searchParams.has('sources')).to.equal(false)
```
[the remark this answers is not in the document]
Comment (ruromero) [reply test/analysis.test.js:241]: [sdlc-workflow/verify-pr] Classified as **suggestion** — proposes additional test coverage for recommend param edge cases. CONVENTIONS.md testing conventions specify 82% line coverage but do not require specific edge case patterns for query parameter testing. No sub-task created.

**Comment by ruromero** [comment]
[sdlc-workflow/verify-pr] Re: @sourcery-ai review — Two suggestions classified:

1. **suggestion** — "Use `set` instead of `append`": CONVENTIONS.md does not document a preference, and the codebase uses `append` exclusively. The function is called once per URL creation, so duplicates cannot occur. No sub-task created.

2. **suggestion** — "Extract CLI wiring helper": CONVENTIONS.md does not document a DRY convention for CLI handlers. Each command independently declares its options (e.g., `--html`, `--summary`, `--workspaceDir` are similarly repeated). No sub-task created.

**Comment by ruromero** [comment]
[sdlc-workflow/verify-pr] Re: @a-oren review — Classified as **code change request** — sub-task [TC-5441](https://redhat.atlassian.net/browse/TC-5441) created to address this feedback. The image CLI command is missing --providers/--sources flags while the three other commands have them, creating an inconsistency.

[TC-5441]: https://redhat.atlassian.net/browse/TC-5441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by ruromero** [comment]
## Verification Report for TC-5409 (commit 647bdc1)

| Check | Result | Details |
|-------|--------|---------|
| Review Feedback | WARN | 1 code change request (image CLI missing flags) → [TC-5441](https://redhat.atlassian.net/browse/TC-5441); 3 suggestions kept as-is |
| Root-Cause Investigation | DONE | plan-feature gap — incomplete CLI command enumeration → [TC-5442](https://redhat.atlassian.net/browse/TC-5442) |
| Scope Containment | WARN | All task files present; 2 extra files (test + index.js typedef) are justified additions |
| Diff Size | PASS | 202 additions, 13 deletions across 4 files — proportionate to task scope |
| Commit Traceability | PASS | Single commit references "Implements TC-5409" |
| Sensitive Patterns | PASS | No secrets, credentials, or sensitive data detected |
| CI Status | PASS | All 5 checks pass (lint+test Node 22/24, Sourcery, PR title, commit messages) |
| Acceptance Criteria | PASS | 5/5 criteria met |
| Test Quality | WARN | Repetitive Test Detection: 3 pairs could be parameterized (matches existing project style); Test Documentation: PASS (all 9 tests have JSDoc); Eval Quality: N/A |
| Test Change Classification | ADDITIVE | 134 lines added, 0 removed; 2 new suites with 9 tests |
| Verification Commands | PASS | `npm run lint` clean (0 errors); `npm test` all 25 analysis tests pass |

### Overall: WARN

One code change request from reviewer @a-oren: the `image` CLI command is missing `--providers`/`--sources` flags while the other three commands have them. Sub-task [TC-5441](https://redhat.atlassian.net/browse/TC-5441) created to track the fix. Root-cause traced to plan-feature phase omitting the `image` command from CLI enumeration — [TC-5442](https://redhat.atlassian.net/browse/TC-5442) created for skill improvement.

All acceptance criteria pass. CI green. No security concerns. Tests are additive with good documentation.

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/RHEcosystemAppEng/sdlc-plugins) v0.13.7.*

[TC-5441]: https://redhat.atlassian.net/browse/TC-5441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5442]: https://redhat.atlassian.net/browse/TC-5442?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5441]: https://redhat.atlassian.net/browse/TC-5441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Review by a-oren — APPROVED** [review]
LGTM
